numpy/random/generator.pyx

```diff
@@ -3475,21 +3492,42 @@ cdef class Generator:
 
         # GH10839, ensure double to make tol meaningful
         cov = cov.astype(np.double)
-        (u, s, v) = svd(cov)
+        if method == 'svd':
+            from numpy.dual import svd
+            (u, s, vh) = svd(cov)
+        elif method == 'eigh':
+            from numpy.dual import eigh
+            # could call linalg.svd(hermitian=True), but that calculates a vh we don't need
+            (s, u)  = eigh(cov)
+        else:
+            from numpy.dual import cholesky
+            l = cholesky(cov)
 
-        if check_valid != 'ignore':
+        # make sure check_valid is ignored whe method == 'cholesky'
+        # since the decomposition will have failed if cov is not valid.
+        if check_valid != 'ignore' and method != 'cholesky':
             if check_valid != 'warn' and check_valid != 'raise':
-                raise ValueError("check_valid must equal 'warn', 'raise', or 'ignore'")
-
-            psd = np.allclose(np.dot(v.T * s, v), cov, rtol=tol, atol=tol)
+                raise ValueError(
+                    "check_valid must equal 'warn', 'raise', or 'ignore'")
+            if method == 'svd':
+                psd = np.allclose(np.dot(vh.T * s, vh), cov, rtol=tol, atol=tol)
+            else:
+                psd = not np.any(s < -tol)
             if not psd:
                 if check_valid == 'warn':
                     warnings.warn("covariance is not positive-semidefinite.",
                                   RuntimeWarning)
                 else:
                     raise ValueError("covariance is not positive-semidefinite.")
 
-        x = np.dot(x, np.sqrt(s)[:, None] * v)
+        if method == 'cholesky':
+            _factor = l
+        elif method == 'eigh':
+            _factor = u * np.sqrt(s)
```

Comment:
Since you don't have `s >= 0` above but instead `s > -tol`, you need to ensure s is positive here so that you don't get `NaN`.

Perhaps it's worth using `svd(hermitian=True)` after all, since getting the alternative right is more work.

Reply:
If I use `svd(hermitian=True)` then that means it only computes `s` and not `u`. This forces us to use the `compute_uv=True` keyword which means we now calculate both u and vh for eigh method. Should I got ahead and change this line to use `linalg.svd` instead of `dual.eigh` @eric-wieser ?

Reply:
I think the tests aren't very useful here - the covariance matrices used are symmetric across the antidiagonal. which means a lot of the differences between svd and eigh.

I suppose the other thing to consider is that the outputs don't need to match anyway.